src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -26,6 +27,59 @@ public override SarifLogVersionOne VisitSarifLogVersionOne(SarifLogVersionOne no
             return null;
         }
 
+        public FileData TransformFileDataVersionOne(FileDataVersionOne node)
+        {
+            FileData fileData = null;
+
+            if (node != null)
+            {
+                fileData = new FileData
+                {
+                    Length = node.Length,
+                    MimeType = node.MimeType,
+                    Offset = node.Offset,
+                    ParentKey = node.ParentKey,
+                    Properties = node.Properties                    
+                };
+
+                if (fileData.Properties == null)
+                {
+                    fileData.Properties = new Dictionary<string, SerializedPropertyInfo>();
+                }
+
+                if (node.Uri != null)
+                {
+                    fileData.FileLocation = new FileLocation
+                    {
+                        Uri = node.Uri,
+                        UriBaseId = node.UriBaseId
+                    };
+                }
+
+                if (!string.IsNullOrWhiteSpace(node.Contents))
+                {
+                    fileData.Contents = new FileContent
+                    {
+                        Binary = node.Contents
+                    };
+                }
+
+                if (node.Hashes != null)
+                {
+                    fileData.Hashes = new List<Hash>();
+
+                    foreach (HashVersionOne hash in node.Hashes)
+                    {
+                        fileData.Hashes.Add(TransformHashVersionOne(hash));
+                    }
+                }
+
+                fileData.Tags.UnionWith(node.Tags);
```

Comment:
Why `UnionWith`? `Tags` is guaranteed empty at this point. Why not just assign?

I had this question on a usage of `UnionWith` in a previous review, but I forgot to comment on it.

Reply:
Tags is get-only

Reply:
Oh!